tests/test_custom_ops.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2022 Graphcore Ltd. All rights reserved.
+import os.path
+from tempfile import NamedTemporaryFile
+
+import pytest
+
+from examples_utils.custom_ops.custom_ops_utils import load_custom_ops_lib, get_binary_path
+
+from multiprocessing import Process
+
+
+def create_cpp_file():
+    """Create C++ file to compile. Create new one per test."""
+    # Create empty temp C++ file to compile
+    cpp_code = """
+    // cppimport
+    #include <pybind11/pybind11.h>
+
+    namespace py = pybind11;
+
+    int square(int x) {
+        return x * x;
+    }
+
+    PYBIND11_MODULE(somecode, m) {
+        m.def("square", &square);
+    }
+    /*
+    <%
+    setup_pybind11(cfg)
+    %>
+    */
+    """
+    file = NamedTemporaryFile(suffix='.cpp', mode='w')
+    file.write(cpp_code)
+    return file
+
+
+def test_custom_ops():
+    cpp_file = create_cpp_file()
+
+    # Compile first time
+    load_custom_ops_lib(cpp_file.name)
+
+    binary_path = get_binary_path(cpp_file.name)
+    assert os.path.exists(binary_path)
+    assert not os.path.exists(binary_path + '.lock')
+
+    # Compile again
+    load_custom_ops_lib(cpp_file.name)
```

Comment:
Why do you need to compile again here?

Reply:
There is no assertion after the second load, is it necessary?

Reply:
Its testing if it can load the op even if its already been compiled. Ive added a comment to make it clearer